ci/test-flow-collector.sh

```diff
@@ -0,0 +1,66 @@
+set -eu
+
+LOGSTASH_PORT=4739
+LOGSTASH_IP="0.0.0.0"
+LOGSTASH_PROTOCOL="udp"
+FAILURE=false
+
+setup_flow_collector() {
+  echo "=== Starting Flow Collector ==="
+  kubectl create namespace antrea-flow-collector
+  kubectl create configmap logstash-configmap -n antrea-flow-collector --from-file=${GIT_CHECKOUT_DIR}/build/yamls/flow-collector/logstash/
+  kubectl apply -f ${GIT_CHECKOUT_DIR}/build/yamls/flow-collector/flow-collector.yml -n antrea-flow-collector
+  echo "=== Waiting for Elastic Stack to be ready ==="
+  kubectl wait --for=condition=ready pod -l app=elasticsearch -n antrea-flow-collector --timeout=600s
+  kubectl wait --for=condition=ready pod -l app=kibana -n antrea-flow-collector
+  kubectl wait --for=condition=ready pod -l app=logstash -n antrea-flow-collector --timeout=600s
+  # get cluster-ip of logstash
+  LOGSTASH_IP=$(kubectl get svc logstash -n antrea-flow-collector -o jsonpath='{.spec.clusterIP}')
+  if [ ${LOGSTASH_PROTOCOL} = "udp" ]; then
+    nc -zvu ${LOGSTASH_IP} ${LOGSTASH_PORT}
+  fi
+  echo "=== Flow Collector is listening on ${LOGSTASH_IP}:${LOGSTASH_PORT} ==="
+}
+
+config_antrea() {
+  echo "=== Stopping Antrea === "
+  kubectl delete -f ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
+  echo "=== Configuring Antrea Flow Exporter Address ==="
+  sed -i -e "s/#flowCollectorAddr.*/flowCollectorAddr: \"${LOGSTASH_IP}:${LOGSTASH_PORT}:${LOGSTASH_PROTOCOL}\"/g" ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
+  sed -i -e "s/#  FlowExporter: false/  FlowExporter: true/g" ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
+  sed -i -e "s/#  AntreaProxy: false/  AntreaProxy: true/g" ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
+  echo "=== Restarting Antrea ==="
+  kubectl apply -f ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
+  sleep 20
+  kubectl wait --for=condition=ready pod -l app=antrea -n kube-system --timeout=300s
+  echo "=== Antrea flow exporter starts to send data ==="
+}
+
+check_record() {
+  echo "=== Wait for 5 minutes to receive data ==="
```

Comment:
since we are not deploying any workload Pods, are we just looking at CoreDNS traffic here?

Reply:
Yes as long as we receive the flow records having one of desired fields I will consider it successful. Do you suggest to add workload pods?

Reply:
If the job works, it's goo enough for me. The whole point of this job is to validate the yaml manifest. May want to add a comment about this though.

Reply:
Sounds good. Done